ci/docker-run.sh

```diff
@@ -96,4 +109,28 @@ fi
 
 set -x
 # shellcheck disable=SC2086
-exec docker run "${ARGS[@]}" $CODECOV_ENVS "$IMAGE" "$@"
+if docker run "${ARGS[@]}" $CODECOV_ENVS "$IMAGE" "$@"; then
+  docker_status=0
+else
+  docker_status=$?
+fi
+
+if [[ -n $CI_WITH_DEPENDABOT && $(git status --short :**/Cargo.lock | wc -l) -gt 0 ]]; then
+  (
+    echo --- "(FAILING) Backpropagating dependabot-triggered Cargo.lock updates"
+    set -x
+
+    git add :**/Cargo.lock
+    NAME="dependabot-buildkite"
+    GIT_AUTHOR_NAME="$NAME" \
+      GIT_COMMITTER_NAME="$NAME" \
+      EMAIL="dependabot-buildkite@noreply.solana.com" \
+      git commit -m "Update all Cargo lock files"
+    git push origin "HEAD:$branch"
```

Comment:
To make it work, this is escaped from more proper place like `test-checks.sh` (maybe comment-in-source so)